src/main/java/com/github/jparkie/pdd/BitArray.java

```diff
@@ -38,22 +38,35 @@ private static int numWords(long numBits) {
         return (int) numWords;
     }
 
+    private final class indexAndMask {
+        public final int index;
+        public final long mask;
+
+        public indexAndMask(long index) {
+            this.index = (int)(index >>> 6);
+            this.mask = 1L << (index & 0x3F);
+        }
+    }
+
     public boolean get(long index) {
-        return (data[(int) (index >>> 6)] & (1L << index)) != 0;
+        final indexAndMask iam = new indexAndMask(index);
```

Comment:
I'm not a huge fan of allocating a new object to the heap every time `get()` is invoked. For the real-time streaming scenario, arbitrary heap allocations in local scope does impact real-time performance. To minimize such an impact would be to rely on an ObjectPool which I do not want to introduce given the optimization the JVM will perform after it warms up in comparison.